app/admin/contributions.rb

```diff
@@ -15,14 +15,25 @@
   member_action :refuse, method: :put, only: :index
 
   batch_action :refuse, if: proc { params[:scope] != "recusado" && params[:scope] != "aprovado" } do |ids|
-    batch_action_collection.find(ids).each { |contribution| contribution.refuse!(current_user.id) if contribution.state == "received" }
+    batch_action_collection.find(ids).each do |contribution|
+      next unless contribution.received?
 
-    redirect_back fallback_location: collection_path, notice: "The contributions have been refused."
+      begin
+        Contribution.transaction do
+          contribution.refuse!(current_user.id)
+        end
```

Comment:
Where did the `update` method go? Since we only have one operation now I think we can get off of this `transaction`.

Reply:
Btw, it's already fixed :+1:

Reply:
cool